### PR TITLE
documentation updates

### PR DESCRIPTION
**Pull request by hannesm**
also, remove the Mirage_flow.stats which isn't used much...

/cc @reynir @dinosaure

### CODE REVIEW

**Comment by hannesm** [comment]
any chance I can get a review and merge and release (//cc @reynir)?

File: src/mirage_flow.mli [hunk 1]
```diff
@@ -50,63 +50,66 @@ module type S = sig
...
+  (** [write flow buffer] writes a buffer to the flow. There is no indication
+      when the buffer has actually been sent and, therefore, it must not be
+      reused. The contents may be transmitted in separate packets, depending on
+      the underlying transport. The result [Ok ()] indicates success,
+      [Error `Closed] indicates that the connection is now closed and therefore
+      the data could not be written. Other errors are possible.
+
+      The call to [write] blocks until the buffer has been accepted by the
+      implementation (if a partial write occured, [write] will wait until the
+      remainder of the buffer has been accepted by the implementation).
```
Comment (reynir) [inline src/mirage_flow.mli]: Hmm, does the *call* block or the *lwt promise* block? I think it is the latter, but the phrasing suggests the former.

File: src/mirage_flow.mli [hunk 2]
```diff
@@ -50,63 +50,66 @@ module type S = sig
...
-      written.  Other errors are possible. *)
+  (** [writev flow buffers] writes a sequence of buffers to the flow. There is
+      no indication when the buffers have actually been sent and, therefore,
+      they must not be reused. The result [Ok ()] indicates success,
+      [Error `Closed] indicates that the connection is now closed and therefore
+      the data could not be written. Other errors are possible.
+
+      The call to [writev] blocks until the buffers have been accepted by the
+      implementation (if a partial write occured, [writev] will wait until all
+      buffers have been accepted by the implementation).
```
Comment (reynir) [inline src/mirage_flow.mli]: Here again call vs. promise.

File: src/mirage_flow.mli [hunk 3]
```diff
@@ -50,63 +50,66 @@ module type S = sig
...
-  write_bytes: int64;
-  write_ops: int64;
-  duration: int64;
-}
-(** The type for I/O statistics from a copy operation. *)
-
-val pp_stats: stats Fmt.t
-(** [pp_stats] is the pretty-printer for flow stats. *)
+      If this [flow] is layered upon another [flow'] (e.g. TLS over TCP),
+      [close] on the underlying [flow'] is executed. *)
```
Comment (reynir) [inline src/mirage_flow.mli:118]: Good

**Review by reynir — APPROVED** [review]

**Pull request merged**